systems/plants/RigidBodyFlatTerrain.m

```diff
@@ -3,13 +3,18 @@
 %  everywhere
   
   methods 
-    function obj = RigidBodyFlatTerrain()
+    function obj = RigidBodyFlatTerrain(varargin)
+      % Construct a flat terrain map at a given z height (default 0)
+      p = inputParser();
```

Comment:
while I agree in principle that we shouldn't be doing argument parsing in the inner-loop, the arguments available to forwardKin have been getting ridiculous.  We've been moving to an options structure for that method.  If you guys think that's the wrong decision, better to pipe up now.

On Feb 17, 2015, at 7:18 AM, Andres Valenzuela <notifications@github.com> wrote:

> In systems/plants/RigidBodyFlatTerrain.m:
> 
> > @@ -3,13 +3,18 @@
> >  %  everywhere
> >    
> >    methods 
> > -    function obj = RigidBodyFlatTerrain()
> > +    function obj = RigidBodyFlatTerrain(varargin)
> > +      % Construct a flat terrain map at a given z height (default 0)
> > +      p = inputParser();
> Looks like inputParser is slower for checking optional inputs, but I haven't found anything on the kind of field checks that we do on our options structs. I'm guessing that we wont want to use it in 'inner-loop' code, but I'd argue that we probably shouldn't be parsing options structs there either.
> 
> I tried converting a couple of functions over to the inputParser style last night - I think I like it better.
> 
> —
> Reply to this email directly or view it on GitHub.
>